client/client.go

```diff
@@ -22,12 +22,12 @@ import (
 
 // Client provides the interface for the consuming application
 type Client struct {
-	engine              engine.Engine // The core business logic of the client
+	Engine              engine.Engine // The core business logic of the client
```

Comment:
Is this intended to be a long term approach or just some quick code to get the prototype working?

My 2c: the client intentionally abstracts the engine from api consumers. An rpc api is a way for remote consumers of a nitro node to call client methods. So the nitro node rpc api code would do the following:
1. Receives some data string from the network.
2. Parses the data string into a jsonrpc request.
3. Invokes a client method based on the jsonrpc request.

Reply:
yes this was a fast approach to get it to work.
You suggest to implement methods at the client level to forward the data from/to the engine?

Reply:
The way I have been thinking about RPC design is:
1. RPC code receives a message/request.
2. RPC code deserializes the message.
3. RPC code calls a client method based on the message.